src/main/java/com/duckblade/osrs/toa/TombsOfAmascutConfig.java

```diff
@@ -562,16 +561,44 @@ default Color scabarasSequenceColorEnd()
 		return Color.blue;
 	}
 
+	String KEY_SCABARAS_MATCHING_DISPLAY_MODE_TILE = "scabarasMatchingDisplayModeTile";
+
 	@ConfigItem(
-		keyName = "scabarasMatchingDisplayMode",
-		name = "Matching Display",
-		description = "Whether to show highlight tiles, show names of tiles, or both for the matching puzzle.",
+		keyName = KEY_SCABARAS_MATCHING_DISPLAY_MODE_TILE,
+		name = "Matching Display - Tile",
+		description = "Whether to highlight tiles in the matching puzzle.",
 		position = 7,
 		section = SECTION_SCABARAS
 	)
-	default MatchingTileDisplayMode scabarasMatchingDisplayMode()
+	default boolean scabarasMatchingDisplayModeTile()
+	{
+		return true;
+	}
+
+	String KEY_SCABARAS_MATCHING_DISPLAY_MODE_NAME = "scabarasMatchingDisplayModeName";
+
+	@ConfigItem(
+		keyName = KEY_SCABARAS_MATCHING_DISPLAY_MODE_NAME,
+		name = "Matching Display - Name",
+		description = "Whether to show the tile name in the matching puzzle.",
+		position = 8,
+		section = SECTION_SCABARAS
+	)
+	default boolean scabarasMatchingDisplayModeName()
 	{
-		return MatchingTileDisplayMode.BOTH;
+		return true;
+	}
+
+	@ConfigItem(
+		keyName = "scabarasMatchingDisplayModeNumber",
+		name = "Matching Display - Number",
+		description = "Whether to show the tile number in the matching puzzle.",
+		position = 9,
+		section = SECTION_SCABARAS
+	)
+	default boolean scabarasMatchingDisplayModeNumber()
+	{
+		return false;
```

Comment:
rather than start adding enum entries for all the possible permutations of things (and then miss out on potentially name + number combos), I replaced each with a boolean.